GitUI/BranchTreePanel/RepoObjectsTree.ContextActions.cs

```diff
@@ -37,91 +37,85 @@ partial class RepoObjectsTree : IMenuItemFactory
         /// </summary>
         private MenuItemsGenerator<TagNode> _tagNodeMenuItems;
 
-        private void ContextMenuAddExpandCollapseTree(ContextMenuStrip contextMenu)
+        /// <summary>Toggles the <paramref name="items"/> <see cref="ToolStripItem.Visible"/>
+        /// as well as <see cref="ToolStripItem.Enabled"/> properties depending on <paramref name="enabled"/>.
+        /// Prefer this over only toggling the visibility of an item to enable determining whether the context menu will (once open)
+        /// contain any visible items via <see cref="ToolStripItem.Enabled"/> in <see cref="contextMenu_Opening(object, CancelEventArgs)"/>
+        /// even before the menu itself (as the visual parent) is visble and <see cref="ToolStripItem.Visible"/> of any item therefore false.</summary>
+        private void ToggleMenuItems(bool enabled, params ToolStripItem[] items) => items.ForEach(item => item.Visible = item.Enabled = enabled);
```

Comment:
@mstv I've applied the formatting, but personally find `SetMenuItems` confusing because they're not set on anything. I understand you associate *toggle* with *checked*. I associate it with *visible*, like in https://api.jquery.com/toggle/ - especially when applied to menu items, where *checking* doesn't make a lot of sense.
Unless we find a more descriptive verb than *toggle* for toggling the `Visible` and `Enabled` flags, I'd say devs will just have to RTFM (i.e. the XML docs I wrote) and learn what *toggle* means in this context. Is that acceptable for you?

Reply:
With the below changes, there might be no use of `params` anymore.
The menu item states `Enabled` and `Visible` _are set_, not toggled (in the meaning used in other GE code).
`EnableMenuItem`?

Reply:
> With the below changes, there might be no use of `params` anymore. The menu item states `Enabled` and `Visible` _are set_, not toggled (in the meaning used in other GE code). `EnableMenuItem`?

@mstv Ah, now I understand your reasoning. You mean the `Enabled` and `Visible` properties of the menu items are being _set_ to the value of the `enabled` parameter instead of being _toggled_ from their current value like `Enabled = !Enabled`.

I meant the method to be understood as toggling (i.e. displaying or hiding) menu items in the context menu as the implicit context of the method. I was aiming for a semantic like in this overload: https://api.jquery.com/toggle/#toggle-display

`EnableMenuItem` (i. e. renaming all methods in the context menu to _Enable..._ instead of _Toggle..._) is fine with me though. I personally prefer the syntax of `toggle(false)` over `enable(false)`, but see how _toggle_ can be misinterpreted if you don't have above semantic in mind. Let me know, if you'd like me to rename the `Toggle...` methods.

Reply:
Yes, please. (Initially, I hesitated to propose the name `EnableMenuItem` because `Visible` is set, too.)